clang/test/CXX/expr/expr.prim/expr.prim.general/p3-0x.cpp

```diff
@@ -113,8 +113,9 @@ namespace Static {
     static auto g() -> decltype(this->m); // expected-error{{'this' cannot be used in a static member function declaration}}
 
     static int h();
-    
-    static int i() noexcept(noexcept(m + 2)); // expected-error{{'this' cannot be implicitly used in a static member function declaration}}
+
+    // The use of 'm' doesn't constitute an ODR use, so we don't have a reason to reject it.
+    static int i() noexcept(noexcept(m + 2));
```

Comment:
All major compilers (except for us) accept the code: https://gcc.godbolt.org/z/ffxK4aKTa

Reply:
I would be great to find the corresponding wording

Reply:
Done. I didn't even know we had relevant wordings about such scenarios until @frederick-vs-ja (thanks!) told me about [[expr.prim.id.general](https://eel.is/c++draft/expr.prim#id.general-4.3)]/4.